crates/miden-objects/src/transaction/proven_tx.rs

```diff
@@ -34,6 +34,12 @@ pub struct ProvenTransaction {
     /// while for public notes this will also contain full note details.
     output_notes: OutputNotes,
 
+    /// [`BlockNumber`] of the transaction's reference block.
+    ///
+    /// This is not needed for proving the transaction, but it is useful for the node to lookup the
+    /// block.
+    block_num: BlockNumber,
```

Comment:
As mentioned in the `miden-node` PR review, I think we should also include block number as an input into the transaction kernel. The idea is to make it so that verifying the proof would guarantee internal consistency of the `ProvenTransaction` struct (we are not quite there yet because the proof currently does not cover the `details` field of the `TxAccountUpdate` - but that's a separate issue).

So, maybe the transaction kernel inputs could look like so:
```
[
    BLOCK_HASH,
    block_num, 0, account_id_prefix, account_id_suffix,
    INITIAL_ACCOUNT_HASH,
    INPUT_NOTES_COMMITMENT,
]
```
Then, verifying that `block_num` is consistent with `BLOCK_HASH` should be just a few extra cycles in the kernel.

Let's create a separate issue to address this in the future.